src/main/scala/io/qbeast/core/model/IndexFile.scala

```diff
@@ -34,7 +34,8 @@ case class IndexFile(
     size: Long,
     modificationTime: Long,
     revisionId: RevisionID,
-    blocks: IISeq[Block])
+    blocks: IISeq[Block],
+    stats: String = null)
```

Comment:
Two things. First, it is best to avoid using null; use Options instead.
Second, I'm not sure about using stats as a String, as I'm not sure that the way we serialize Stats into a string is the same in Delta/Iceberg or Hudi.

If this is not the case (the 3 formats use different ways to store metadata), I think it would be best to create our own stats representation (some case classes that follow our naming convention) and then define specific adapters for each format.

Reply:
I fixed this (avoid using null), but for now I keep it as a string.
I will create a new QbeastStats representation in a separate PR